state/watcher_profilecharm_test.go

```diff
@@ -206,12 +206,12 @@ func (s *instanceCharmProfileWatcherSuite) expectLoopCollectionFilterAndNotify(c
 	}
 }
 
-func (s *instanceCharmProfileWatcherSuite) expectMergeCollectionInstanceField(state string) func() {
+func (s *instanceCharmProfileWatcherSuite) expectMergeCollectionInstanceField(value string) func() {
 	return func() {
 		s.database.EXPECT().GetCollection("instanceCharmProfileData").Return(s.collection, noop)
 		s.collection.EXPECT().Find(bson.D{{"_id", "1"}}).Return(s.query)
-		s.query.EXPECT().One(gomock.Any()).SetArg(0, map[string]interface{}{
-			"upgradecharmprofilecomplete": state,
+		s.query.EXPECT().One(gomock.Any()).SetArg(0, state.InstanceCharmProfileDataDoc{
+			UpgradeCharmProfileComplete: value,
```

Comment:
This makes it worth it, as we no longer have to guess what the string value should be, esp. if somebody changes the name of the field.